Classes/Controller/AbstractApiController.php

```diff
@@ -47,34 +47,39 @@ public function injectLogger(\PunktDe\PtExtbase\Logger\Logger $logger)
     }
 
 
-
     /**
-     * Handles all exceptions thrown inside the application
+     * @return int|mixed
+     * @throws Exception
+     * @throws \TYPO3\CMS\Extbase\Mvc\Exception\StopActionException
      */
     protected function callActionMethod()
     {
         try {
             parent::callActionMethod();
         } catch (Exception $exception) {
+
+
             if ($exception instanceof \TYPO3\CMS\Extbase\Mvc\Exception\StopActionException) {
                 throw $exception;
             }
+            $this->cleanUpAtException($exception);
+            $this->objectManager->get('TYPO3\CMS\Extbase\Persistence\Generic\PersistenceManager')->persistAll();
+
+            if (!class_exists(Raven_ErrorHandler::class)) {
```

Comment:
In einer generischen Extension auf das Vorhandensein einer Erweiterung zu prüfen sollte wirklich Ultima Ratio sein. Was wenn jemand ein anderes Exception Tracking  verwenden will? Ausserdem verhält es sich in dem Fall nun anders und gibt nicht mehr den Exception Code alleine raus.